File: iOS/Classes/DoricPlatformAPILibrary.m
#import "DoricPlatformAPILibrary.h"
#import "DoricClipboardPlugin.h"
#import "DoricKeepScreenOnPlugin.h"

@implementation DoricPlatformAPILibrary
- (void)load:(DoricRegistry *)registry {
    NSString *path = [[NSBundle mainBundle] bundlePath];
    NSString *fullPath = [path stringByAppendingPathComponent:@"bundle_doric-platform-api.js"];
    NSString *jsContent = [NSString stringWithContentsOfFile:fullPath encoding:NSUTF8StringEncoding error:nil];
    [registry registerJSBundle:jsContent withName:@"doric-platform-api"];
    [registry registerNativePlugin:DoricClipboardPlugin.class withName:@"clipboard"];
    [registry registerNativePlugin:DoricKeepScreenOnPlugin.class withName:@"screenKeepOn"];
}
@end

File: android/src/main/java/pub/doric/library/DoricPlatformAPILibrary.java
package pub.doric.library;

import java.io.IOException;
import java.io.InputStream;

import pub.doric.Doric;
import pub.doric.DoricComponent;
import pub.doric.DoricLibrary;
import pub.doric.DoricRegistry;

@DoricComponent
public class DoricPlatformAPILibrary extends DoricLibrary {
    @Override
    public void load(DoricRegistry registry) {
        try {
            InputStream is = Doric.application().getAssets().open("bundle_doric-platform-api.js");
            byte[] bytes = new byte[is.available()];
            is.read(bytes);
            String content = new String(bytes);
            registry.registerJSBundle("doric-platform-api", content);
        } catch (IOException e) {
            e.printStackTrace();
        }
        registry.registerNativePlugin(DoricScreenKeepOnPlugin.class);
        registry.registerNativePlugin(DoricClipboardPlugin.class);
    }
}
